### DISCUSSION TITLE
failed optimization at low mach number 0.044 at maximization lift for single point optimization for same tutorial NACA0012

### DISCUSSION THREAD

**Discussion by EngJamal2023** [mdolab/MACH-Aero#125] [Q&A] [answered]
I kindly request assistance in achieving a favorable outcome for my single point optimization case. Despite dedicating an entire month to identifying my mistakes, I have been unable to pinpoint the error I made. I am hopeful that you can provide me with guidance. To this end, I have made modifications to the opt_airfoil.py code as outlined below:

![image](https://github.com/mdolab/MACH-Aero/assets/137990941/539eeb20-a315-45ea-bb4b-afd73c397e7e)
and i have been changed on the objective function and the sensitivity function as following ;
![image](https://github.com/mdolab/MACH-Aero/assets/137990941/69c660c0-222e-49c9-baa9-0dd1eeeb2751)
and also the drag constraint as following 👍 
![image](https://github.com/mdolab/MACH-Aero/assets/137990941/a37943f8-21d8-4284-aa2b-d41ba58e83ce)

I sincerely hope to receive assistance and guidance regarding my current issues. It would be greatly appreciated if you could suggest a guide book or provide tutorials on Python scripting for single point optimization and other optimization techniques. Despite my extensive reading, I have been unable to find the answers to my questions. Therefore, I kindly request your help in identifying the possible changes that I should focus on when conducting an analysis on different airfoils to address the issue of solution divergence. Your support in this matter would be invaluable to me.

**Comment by hajdik** [comment, marked as answer]
If you are maximizing a function, I believe you only need to negate it in `objCon`. This will already affect the derivatives so you do not need to negate them in `cruiseFuncSens` too.

**Reply by EngJamal2023** [reply]
thank you so much for your reply and solve the problem to me i appreciate it
